### PR TITLE
Adds search specific, merge search-moar into master, resolves [User Story 5] #176

### PR DESCRIPTION
**Pull request by kevcamel**
Added further search functioanlity. Categories filter now available on search.resolves [User Story 5] #176.

### CODE REVIEW

File: app/Http/Controllers/searchController.php [hunk 1]
```diff
@@ -16,6 +16,23 @@ public function search($searchKey)
     {
 		$products = Product::search($searchKey)->get();
 		$searchWord = '$searchKey';
-		return view('search',['searchWord'=>$searchKey], compact('products'));
+		return view(
+			'search',
+			[
+				'searchWord'=>$searchKey
+			],
+			compact('products'));
```
Comment (BrandonBalala) [inline app/Http/Controllers/searchController.php:24]: what does compact('products') do?

File: resources/views/search.blade.php [hunk 2]
```diff
@@ -2,42 +2,122 @@
...
-    				<p class="card-text mb-2">{{ $product->description }}</p>
-    				<footer class="text-right">
-    					<small class="text-muted">{{ date('M-jS', strtotime($product->created_at)) }}</small><br/>
-    					<a href="/categories/{{$product->category->id}}"><small class="badge badge-pill badge-info">{{ $product->category->name}}</small></a>
-    					<small class="badge badge-pill badge-success">$ {{ $product->price }}</small>
-    				</footer>
-    			</div>
-    		</div>
-    		<!-- End Card -->
+            @if( isset($category) && ( ($product->category->name == $category->name) || (isset($product->category->parent) && $product->category->parent->name == $category->name) || (isset($product->category->parent->parent) && $product->category->parent->parent->name == $category->name) ) )
```
Comment (BrandonBalala) [inline resources/views/search.blade.php:85]: This line is super messy, maybe you should make an array of all the category ids in the controller and pass it to the view and then simply use in_array.

http://php.net/manual/en/function.in-array.php

File: app/Http/Controllers/searchController.php (hunk 1 above)
Comment (monnyy) [reply app/Http/Controllers/searchController.php:24]: It creates an array of variables and their values. In this case, value stored in variable 'product's which is created in line 17 will be returned (displayed)

Comment (kevcamel) [reply app/Http/Controllers/searchController.php:24]: it creates an array of the results 'products'

File: resources/views/search.blade.php (hunk 2 above)
Comment (kevcamel) [reply resources/views/search.blade.php:85]: hmm... i agree. However it is functional and there is alot of work to do. I will create an issue and resolve it in the next sprint.

Comment (BrandonBalala) [reply resources/views/search.blade.php:85]: alright

**Pull request merged**